docs/adr/0006-process-for-onboarding-new-tenants.md

```diff
@@ -0,0 +1,270 @@
+
+As with Workflow, the first thing that happens is that the Operations service will attach it's storage configuration to the Litware tenant, and then create a sub-tenant of it's own Service Tenant with which it will call it's dependencies:
+
+```
+Root tenant
+ |
+ +-> Contoso
+ |
+ +-> Litware
+ |     +-> (Workflow storage configuration)
+ |     +-> (The Id of the WORKFLOW+Litware sub-tenant for the Workflow service to use)
+ |     +-> (Operations storage configuration)
+ |     +-> (The Id of the OPERATIONS+Litware sub-tenant for the Operations service to use)
+ |
+ +-> WORKFLOW
+ |     |
+ |     +-> WORKFLOW+Litware
+ |           +-> (Operations storage configuration)
+ |           +-> (The Id of the OPERATIONS+WORKFLOW+Litware sub-tenant for the Operations service to use)
+ |           +-> (FooBar storage configuration)
+ |
+ +-> OPERATIONS
+ |     |
+ |     +-> OPERATIONS+WORKFLOW+Litware
+ |     |     +-> (FooBar storage configuration)
+ |     |
+ |     +-> OPERATIONS+Litware
+ |
+ +-> FOOBAR
+```
+
+Then, the Operations service will call out to the FooBar service to enroll the OPERATIONS+Litware service. As with enrolling the OPERATIONS+WORKFLOW+Litware service, this results in FooBar storage configuration being attached to the OPERATIONS+Litware service:
+
+```
+Root tenant
+ |
+ +-> Contoso
+ |
+ +-> Litware
+ |     +-> (Workflow storage configuration)
+ |     +-> (The Id of the WORKFLOW+Litware sub-tenant for the Workflow service to use)
+ |     +-> (Operations storage configuration)
+ |     +-> (The Id of the OPERATIONS+Litware sub-tenant for the Operations service to use)
+ |
+ +-> WORKFLOW
+ |     |
+ |     +-> WORKFLOW+Litware
+ |           +-> (Operations storage configuration)
+ |           +-> (The Id of the OPERATIONS+WORKFLOW+Litware sub-tenant for the Operations service to use)
+ |           +-> (FooBar storage configuration)
+ |
+ +-> OPERATIONS
+ |     |
+ |     +-> OPERATIONS+WORKFLOW+Litware
+ |     |     +-> (FooBar storage configuration)
+ |     |
+ |     +-> OPERATIONS+Litware
+ |           +-> (FooBar storage configuration)
+ |
+ +-> FOOBAR
+```
+
+This completes the enrollment of Litware to the Workflow and Operations services. As can be seen from the above, there are three different paths through which Litware make indirect use of the FooBar service, and it's possible for the client to use separate storage for each. In fact, this will be the default; even if the client is using Marain storage, the data for their three different usage scenarios for FooBar will be stored in different containers.
```

Comment:
I think it is important to make clear that the Client is *unaware* of (though may be able infer) the use of the FooBar service in two of those cases as they are an implementation detail of the services on which the client depends. You mention "is able to access those services" (or words to that effect) in a paragraph above and that is incorrect; "indirectly makes use of" is more accurate.

Reply:
I've added some additional wordage which hopefully reinforces that point.